src/plugins/ImportArtifact/ImportArtifact.js

```diff
@@ -2,81 +2,57 @@
-     * @class
-     * @augments {PluginBase}
-     * @classdesc This class represents the plugin ImportArtifact.
-     * @constructor
-     */
-    var ImportArtifact = function () {
+    const ImportArtifact = function () {
         // Call base class' constructor.
         PluginBase.call(this);
         this.pluginMetadata = pluginMetadata;
     };
 
-    /**
-     * Metadata associated with the plugin. Contains id, name, version, description, icon, configStructue etc.
-     * This is also available at the instance at this.pluginMetadata.
-     * @type {object}
-     */
     ImportArtifact.metadata = pluginMetadata;
 
     // Prototypical inheritance from PluginBase.
     ImportArtifact.prototype = Object.create(PluginBase.prototype);
+    Object.assign(ImportArtifact.prototype, Artifacts.prototype);
+
     ImportArtifact.prototype.constructor = ImportArtifact;
 
-    /**
-     * Main function for the plugin to execute. This will perform the execution.
-     * Notes:
-     * - Always log with the provided logger.[error,warning,info,debug].
-     * - Do NOT put any user interaction logic UI, etc. inside this method.
-     * - callback always has to be called even if error happened.
-     *
-     * @param {function(string, plugin.PluginResult)} callback - the result callback
-     */
     ImportArtifact.prototype.main = async function (callback) {
         const config = this.getCurrentConfig();
-        const hash = config.dataHash;
+        const base = this.getBaseNode();
+        const path = config.dataPath;
         const baseName = config.dataTypeId;
-        const metaDict = this.core.getAllMetaNodes(this.activeNode);
-        const metanodes = Object.keys(metaDict).map(id => metaDict[id]);
-        const base = metanodes.find(node =>
-            this.core.getAttribute(node, 'name') === 'Data'
-        );
 
         if (!base) {
             callback(`Could not find data type "${baseName}"`, this.result);
             return;
         }
 
-        // Get the base node
         const parent = await this.getArtifactsDir();
         const dataNode = this.core.createNode({base, parent});
 
-        const name = await this.getAssetName(hash) ||
+        const name = await this.getAssetNameFromPath(path) ||
             baseName[0].toLowerCase() + baseName.substring(1);
-
-        const assetInfo = await this.transfer(hash, config.storage, name);
-        this.core.setAttribute(dataNode, 'data', JSON.stringify(assetInfo));
-        this.core.setAttribute(dataNode, 'type', baseName);
-        this.core.setAttribute(dataNode, 'createdAt', Date.now());
+        let assetInfo;
 
         try {
-            this.core.setAttribute(dataNode, 'name', name);
-            await this.save(`Uploaded "${name}" data`);
+
+            assetInfo = await this.symLink(path, config.storage);
+            this.assignAssetAttributes(dataNode, {data: assetInfo, name: name, type: baseName});
+            await this.save(`Transferred "${name}" data`);
```

Comment:
So it isn't actually transferring anything...

Reply:
Changed this to `successfully imported ${name} data.`